teps/0115-tekton-catalog-reorganization.md

```diff
@@ -0,0 +1,288 @@
+experiences from [TEP-0003][tep-0003]. In [TEP-0079: Tekton Catalog Support Tiers][tep-0079], we plan to add a Tekton 
+Official Catalog which should use this new organization contract - this proposal is a blocker for [TEP-0079][tep-0079].
+
+## Proposal
+
+We propose supporting a git-based organization of Tekton Catalogs where versions are expressed using commits and tags.
+
+### Flat Structure 
+
+In this organization contract, we'll use a flat structure instead of the directory structure. The example shown above
+with the two `Tasks` (argocd and golang-build) and one `Pipeline` (go-release) would be reorganized as such:
+
+```shell
+./task/
+  /argocd
+    /README.md
+    /argocd.yaml
+    /samples
+    /tests
+    /OWNERS
+  /golang-build
+    /README.md
+    /golang-build.yaml
+    /samples
+    /tests
+    /OWNERS
+./pipeline/
+  /go-release
+    /README.md
+    /go-release.yaml
+    /samples
+    /tests
+    /OWNERS
+```
+
+### Versioning
+
+#### Commits
+
+Users can reference specific versions of the resources using commits. This enables users to pin versions to immutable
+commits, ensuring that the behavior is predictable and reproducible.
+
+##### Git Resolver
+
+[Git Resolver][git-resolver] already has a `"commit"` parameter that can be used to fetch a specific version of a 
+resource, for example [this commit][bp-05] of buildpacks `Task`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  taskRef:
+    resolver: git
+    resource:
+    - name: url
+      value: https://github.com/tekton/catalog.git
+    - name: path
+      value: task/buildpacks/buildpacks.yaml
+    - name: commit
+      value: d9183e36f00c712b5de59932650c071674ea89b8
+```
+
+Note that a user can specify either `"commit"` or `"branch"`, but not both.
+
+##### Hub Resolver
+
+We propose adding `"commit"` parameter to the [Hub Resolver][hub-resolver] that can be used to get a specific version
+of a resource, for example [this commit][bp-05] of buildpacks `Task`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  taskRef:
+    resolver: hub
+    resource:
+    - name: kind
+      value: task
+    - name: name
+      value: buildpacks
+    - name: commit
+      value: d9183e36f00c712b5de59932650c071674ea89b8
+```
+
+Note that a user can specify either `"commit"` or `"version"`, but not both.
+
+#### Tags
+
+Tagged releases should be made so that users who'd prefer to use tags instead of commits can depend on them. The tags
+for versioning Catalogs should continue to use the simpler version of [semver][semver] - `{major}.{minor}` - that's 
+already used in Catalogs today.
+
+TODO(jerop): add more details on tagging
+
+##### Git Resolver
+
+[Git Resolver][git-resolver] has a `"branch"` parameter that can be used to get a tagged version of a resource,
+for example [this commit][bp-05] of buildpacks `Task` is tagged with "v0.5":
```

Comment:
nit: we should make sure we are constistent here. In `git-init`, we do use a `revision` flag/parameters, and I think we should do it there as well. Revision in `git` can be a tag, branch, or a commit as well.

Reply:
agreed, updated the proposal to say that we should replace "commit" and "branch" in git-resolver with "revision" which can take commit, branch or tags - is that what you had in mind @vdemeester?

Reply:
@jerop yep, that's right 👼🏼